plugins/gpg-agent/gpg-agent.plugin.zsh

```diff
@@ -1,14 +1,17 @@
-# Enable gpg-agent if it is not running
-GPG_AGENT_SOCKET="${XDG_RUNTIME_DIR}/gnupg/S.gpg-agent.ssh"
-if [ ! -S $GPG_AGENT_SOCKET ]; then
-  gpg-agent --daemon >/dev/null 2>&1
-  export GPG_TTY=$(tty)
+# Enable gpg-agent if it is not running-
+# --use-standard-socket will work from version 2 upwards
+
+AGENT_SOCK=`gpgconf --list-dirs | grep agent-socket | cut -d : -f 2`
+
+if [ ! -S ${AGENT_SOCK} ]; then
+  gpg-agent --daemon --use-standard-socket >/dev/null 2>&1
 fi
+export GPG_TTY=$(tty)
 
-# Set SSH to use gpg-agent if it is configured to do so
-GNUPGCONFIG="${GNUPGHOME:-"$HOME/.gnupg"}/gpg-agent.conf"
-if [ -r "$GNUPGCONFIG" ] && grep -q enable-ssh-support "$GNUPGCONFIG"; then
+# Set SSH to use gpg-agent if it's enabled
+GPG_SSH_AUTH_SOCK="${AGENT_SOCK}.ssh"
```

Comment:
if you use gpg-conf with list dir, you will have full path for socket.

Reply:
Again, see above. Version 2.0 does not return the ssh path!

I would love if this was simpler, but I have found all these to be necessary. But I do think it is still necessary to support version 2.0 as CentOS 7 as well as Ubuntu 14.04 are both shipping with version 2.0.22 so I don't think this is an uncommon setup?

Reply:
This check is not only related to GPG_SSH_AUTH_SOCK,  you've removed check which is leaving gnupg agent intact if there is no ssh support enabled.

Reply:
The check itself is:

````zsh
    if [ -S ${GPG_SSH_AUTH_SOCK} ]; then
````
So we are checking that the ssh socket exists, which feels to me like a good robust way of checking for support

Version 2.2 of gpg-agent says"The OpenSSH Agent protocol is always enabled,"; so it is possible to use gpg-agent without adding the option to the config file. 

But it is a good question, if someone doesn't want to use gpg-agent as an ssh-agent then we probably shouldn't do the setup. This could either be  by going back to the previous technique of checking the config or possibly by checking that SSH_AUTH_SOCK isn't  already set?